docs/Auth-with-Ionic3-Angular4.md

```diff
@@ -0,0 +1,684 @@
+
+Now, let's try to run the app in browser. Execute the command
+
+```
+
+C:\projects\auth-ng4-ionic3-af2> ionic run android
+
+```
+
+This should run the app on your mobile phone. Now click on the Facebook button and you'll notice the button doesn't work anymore.
+This is because the code written so far is good for running our application in browsers, but when running the application on
+mobile phones, we need to have access to ***Native Mobile API's***, which are provided by _Corodova Plugins_.
+
+**We can access these corodva plugins, using Ionic Native, which are nothing but wrappers for cordova plugins.**
+
+List of all Ionic Native API's for corodova plugins can be found [here](http://ionicframework.com/docs/v2/native/).
+
+Let's look at configuring and installing facebook plugin [here](http://ionicframework.com/docs/v2/native/facebook/).
+_Ensure you follow the steps correctly to configure your app._
+
+Login to facebook dashboard [here](https://developers.facebook.com/apps/), go to your App and make a note of your App ID.
+Next go to command prompt in your project directory and execute the following command by replacing the `APP_ID` with your App Id 
+and `APP_NAME` with your App Name.
+
+ ```bash
+ 
+C:\projects\auth-ng4-ionic3-af2>
+ionic plugin add cordova-plugin-facebook4 --variable APP_ID="123456789" --variable APP_NAME="myApplication" --save
+
+```
+
+This should add following entry in your config.xml, located at the root of your project.
+
+```bash
+
+<plugin name="cordova-plugin-facebook4" spec="~1.7.4">
+        <variable name="APP_ID" value="1689092456387654" />
+        <variable name="APP_NAME" value="auth-ng4-ionic3-af2" />
+</plugin>
+
+```
+This will install the corodva plugin for facebook. 
+
+You should also see a sub-folder named `cordova-plugin-facebook4` under your `plugins` folder.
+
+Add the native dependencies by executing the following command.
+
+```bash
+
+C:\projects\auth-ng4-ionic3-af2>npm install --save @ionic-native/facebook
+
+```
+
+After executing the above command, ensure you got following entry in your `package.json`
+
+```bash
+
+"@ionic-native/facebook": "^3.5.0",
+
+```
+Now import the `Platform` and `Facebook` objects in your `home.ts` and inject the objects in constructor.
+
+```bash
+
+import { Platform } from 'ionic-angular';
+import { Facebook } from 'ionic-native';
+
+```
+
+Update the "signInWithFacebook" method and use Platform Service to determine the platform and run the appropriate code. 
+
+You'll also need to refer firebase in your `home.ts`.
+
+your `home.ts` should look as below
+
+```bash
+
+import { Component } from '@angular/core';
+import { NavController } from 'ionic-angular';
+
+import { AuthProviders, AuthMethods, AngularFire } from 'angularfire2';
+import { Facebook } from '@ionic-native/facebook';
+import { Platform } from 'ionic-angular';
+import * as firebase from 'firebase';
```

Comment:
Wouldn't it be better to just import the things which are needed (for tree-shaking)?

e.g. only import auth 
`import { auth } from 'firebase/app'` and then use `auth.FacebookAuthProvider`